observation_portal/common/schema.py

```diff
@@ -0,0 +1,187 @@
+from rest_framework import serializers
+from rest_framework.schemas.openapi import AutoSchema, SchemaGenerator
+from rest_framework.schemas.utils import is_list_view
+from rest_framework.viewsets import GenericViewSet
+from setuptools_scm import get_version
+from setuptools_scm.version import ScmVersion
+
+
```

Comment:
This file contains our overrides for the base DRF doc generation functionality - I have tried to comment it as much as I can

In a nutshell:
* We can exclude actions in a viewset that we don't want documented
* We can denote a endpoint as having an empty request body - particularly useful for endpoints like "revoke token" or "accept terms"
* If the DRF autodocs are incorrectly interpreting a view as a list view, we can tell it not to via a constructor arg.
* We can also override the request/response examples with our own Python dictionaries
* If a view implements `get_request_serializer()` and/or `get_response_serializer()` the `ObservationPortalSchemaGenerator` class can and will recognize those and generate example requests and responses from them - see comment and related PR for more info.